lib/IDE/ExprContextAnalysis.cpp

```diff
@@ -49,16 +49,20 @@ void swift::ide::typeCheckContextAt(DeclContext *DC, SourceLoc Loc) {
   // Make sure the extension has been bound, in case it is in an inactive #if
   // or something weird like that.
   {
-    SmallVector<ExtensionDecl *, 1> extensions;
-    for (auto typeCtx = DC->getInnermostTypeContext(); typeCtx != nullptr;
-         typeCtx = typeCtx->getParent()->getInnermostTypeContext()) {
-      if (auto *ext = dyn_cast<ExtensionDecl>(typeCtx))
-        extensions.push_back(ext);
-    }
-    while (!extensions.empty()) {
-      extensions.back()->computeExtendedNominal();
-      extensions.pop_back();
-    }
+    class ExtensionBinder: public ASTWalker {
```

Comment:
Why does this traverse the AST instead of iterating over the top level declarations in a file or something?

Reply:
I thought we need to handle something like:
```
struct Something {

extension A: B.#^HERE^# {
}
```
But I didn't know this already has `ErrorType` as the extended type.
Will fix it.

Reply:
> this already has ErrorType as the extended type.

Well, this is not true. But somehow 
```
struct Something {

extension A: B.#^HERE^# {
}
```
doesn't crash. I will investigate.

Reply:
Malformed ASTs makes sense. Thanks.

Reply:
Ah, I see. Non-toplevel `extension` is `canNeverBeBound()`. So we don't need to compute the extended nominal unless it's top-level.